app/models/carto_uploader.rb

```diff
@@ -33,29 +52,92 @@ def upload(data)
       end
       keys = keys.join(",")
       vals = vals.join(",")
-      allVals.push("(#{vals})");  #For Mass Insertion mode only.
+      arrVals.push("(#{vals})")
       #--------------------------------
-
-      #Option 1: Individual SQL Insertions
+      
+      #If we have enough classifications queued up, do a batch Insert.
       #--------------------------------
-      if @mode != CARTODB_INSERT_MODE_MASS
-        sqlQuery = "INSERT INTO #{CARTODB_TABLE} (#{keys}) VALUES (#{vals}) "
-        uri = URI(CARTODB_URI_GET.gsub(/__SQLQUERY__/, URI.escape(sqlQuery)))
-        res = Net::HTTP.get(uri)
-        puts res
+      if (arrVals.length >= CLASSIFICATIONS_PER_BATCH)
+        expectedInsertions += arrVals.length
+        successfulInsertions += batchInsert(keys, arrVals)
+        arrVals = []
+        #Sanity check
+        puts "Inserted #{successfulInsertions} Classifications out of #{expectedInsertions}"
+        if (expectedInsertions != successfulInsertions)
+          #TODO
+        end
       end
       #--------------------------------
     end
 
-    #Option 2: Mass SQL Insertions
+    #Do a final batch Insert for any stragglers
+    #--------------------------------
+    if (arrVals.length > 0)
+      expectedInsertions += arrVals.length
+      successfulInsertions += batchInsert(keys, arrVals)
+      arrVals = []
+    end
+    #--------------------------------
+    
+    #Final Sanity check
     #--------------------------------
-    allVals = allVals.join(",")
-    if @mode == CARTODB_INSERT_MODE_MASS
+    puts "Inserted #{successfulInsertions} Classifications out of #{expectedInsertions}"
+    if (expectedInsertions != successfulInsertions)
+      #TODO
+    end
+    #--------------------------------
+  end
+  
+  #Returns latest Classification ID.
+  def getLatestClassificationID()
+    begin
```

Comment:
If the entire method body is inside a begin/rescue/end you can leave out the begin/end like so:

```ruby
def foo
  raise "hell" if rand > 0.5
rescue StandardError => err
  puts err
end
```